homeassistant/components/spc/__init__.py

```diff
@@ -37,45 +41,75 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS = [Platform.ALARM_CONTROL_PANEL, Platform.BINARY_SENSOR]
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the SPC component."""
 
-    async def async_update_callback(spc_object):
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
```

Comment:
Keep in mind that we need to raise error messages for what could go wrong. If we failed connection or if there was an unknown exception.

This way we let the user know what happened during import in a user friendly way.

Be sure to check Niko Home Control for this

Reply:
Got it! added, raising to ConfigEntryNotReady which triggers a retry.
Don't see a reason to use ConfigEntryError and halt as the pyspcwebgw doesn't provide enough information to know if its an auth/config issue or just a temporary network issue.